nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/lite/ConsumeGCPubSubLite.java

```diff
@@ -0,0 +1,284 @@
+        attributes.put(MSG_PUBLISH_TIME_ATTRIBUTE, String.valueOf(message.getMessage().getPublishTime().getSeconds()));
+        attributes.putAll(message.getMessage().getAttributesMap());
+
+        flowFile = session.putAllAttributes(flowFile, attributes);
+        flowFile = session.write(flowFile, out -> out.write(message.getMessage().getData().toStringUtf8().getBytes()));
+
+        session.transfer(flowFile, REL_SUCCESS);
+        session.getProvenanceReporter().receive(flowFile, context.getProperty(SUBSCRIPTION).evaluateAttributeExpressions().getValue());
+
+        message.getConsumer().ack();
+    }
+
+    private Subscriber getSubscriber(final ProcessContext context) throws IOException {
+
+        final SubscriptionPath subscriptionPath = SubscriptionPath.parse(context.getProperty(SUBSCRIPTION).evaluateAttributeExpressions().getValue());
+
+        final FlowControlSettings flowControlSettings = FlowControlSettings.builder()
+                .setBytesOutstanding(context.getProperty(BYTES_OUTSTANDING).evaluateAttributeExpressions().asDataSize(DataUnit.B).longValue())
+                .setMessagesOutstanding(context.getProperty(MESSAGES_OUTSTANDING).evaluateAttributeExpressions().asLong())
+                .build();
+
+        final MessageReceiver receiver =
+                (PubsubMessage message, AckReplyConsumer consumer) -> {
+                    try {
+                        messages.put(new Message(message, consumer));
+                    } catch (final InterruptedException e) {
+                        getLogger().error("Could not save the message inside the internal queue of the processor", e);
+                    }
+                };
+
+        final SubscriberSettings subscriberSettings = SubscriberSettings.newBuilder()
+                .setCredentialsProvider(FixedCredentialsProvider.create(getGoogleCredentials(context)))
+                .setSubscriptionPath(subscriptionPath)
+                .setReceiver(receiver)
+                .setPerPartitionFlowControlSettings(flowControlSettings)
+                .build();
+
+        return Subscriber.create(subscriberSettings);
+    }
+
+    private class Message {
+        private PubsubMessage message;
+        private AckReplyConsumer consumer;
+
+        public Message(final PubsubMessage message, final AckReplyConsumer consumer) {
+            this.message = message;
+            this.consumer = consumer;
+        }
+
+        public PubsubMessage getMessage() {
+            return message;
+        }
+
+        public AckReplyConsumer getConsumer() {
+            return consumer;
+        }
+    }
+
+    @Override
+    public List<ConfigVerificationResult> verify(ProcessContext context, ComponentLog verificationLogger, Map<String, String> attributes) {
+        final List<ConfigVerificationResult> verificationResults = new ArrayList<>();
+        try {
+            if (subscriber == null) {
+                subscriber = getSubscriber(context);
```

Comment:
Rather than setting the actual instance variable `subscriber`, we should simply call the `getSubscriber` method and catch any exceptions.  We don't want the verification process to actually modify the class.